generator.d

```diff
@@ -803,20 +803,33 @@ void generateAssignmentShared(GeneratorState state, Node target,
         targetRendered = renderLocal(targetLocal);
         targetType = targetLocal.type;
 
-        // Deal with dereference if necessary
+	// Deal with dereference if necessary
         if (targetDereference)
         {
             tempTarget = state.addTemp(targetLocal.type);
             state.render(format("    mov %s, %s", tempTarget.register,
                                 targetRendered));
             targetRendered = format("[%s]", tempTarget.register);
         }
+
+	state.render(format("; %s =", targetBinding.name));
     }
 
+    Type t;
     auto value = generateNode(state, expression);
-    auto valueRendered = renderNode(state, value);
+    auto valueRendered = renderNodeSetType(state, value, &t);
+
+    auto sizeHint = to!string(typeToOpSize(t));
 
-    auto sizeHint = to!string(typeToOpSize(targetType));
+    auto binding = cast(Binding)value;
+    if (binding !is null)
+    {
+        tempTarget = state.addTemp(binding.type);
```

Comment:
Oh I was thinking of making an array of `tempTargets`, but just having a tempTarget2 is even simpler.